### PR TITLE
First round of doc

### PR DESCRIPTION
**Pull request by davidos-b**

### CODE REVIEW

File: include/ardpolyfills/WString.h [hunk 1]
```diff
@@ -83,52 +83,126 @@ class String {
...
+    **/
     [[nodiscard]] bool endsWith(const String& s) const noexcept;
 
+    /**
+    * Copies the String�s characters to the supplied buffer
+    **/
     void getBytes(byte* buffer, unsigned length) const noexcept;
 
+    /**
+    * Locates a character within a string
```
Comment (RuthgerD) [inline include/ardpolyfills/WString.h:107]: Describe what it returns as well

File: include/ardpolyfills/WString.h [hunk 2]
```diff
@@ -83,52 +83,126 @@ class String {
...
+    **/
     [[nodiscard]] float toFloat() const noexcept;
 
+    /**
+    * Get a lower-case version of a String
+    **/
     void toLowerCase() noexcept;
 
+    /**
+    * Get a upper-case version of a String
```
Comment (RuthgerD) [inline include/ardpolyfills/WString.h:187]: Method doesn't return anything, it transforms the string on which this was called to uppercase

File: include/ardpolyfills/Servo.h [hunk 3]
```diff
@@ -25,18 +25,50 @@ enum SMCE__Boundary{ MIN_PULSE_WIDTH = 1000, MAX_PULSE_WIDTH = 2000, DEFAULT_PUL
...
+    **/
     void attach(int pin, int min, int max);
+   
+    /**
+    * Detaches the pin to default value 
+    **/
     void detach();
+    
+    /**
+    * Writes a value that controls the shaft accoddingly
```
[the remark this answers is not in the document]
Comment (davidos-b) [reply include/ardpolyfills/Servo.h:47]: Heh what is broken about it? I dont see it

File: include/ardpolyfills/Servo.h [hunk 4]
```diff
@@ -25,18 +25,50 @@ enum SMCE__Boundary{ MIN_PULSE_WIDTH = 1000, MAX_PULSE_WIDTH = 2000, DEFAULT_PUL
 
 class Servo {
   public:
-    constexpr Servo() = default;
+    /// Default constructor
+    constexpr Servo() = default; 
+   
+    /**
+    * Attaches a pin to attached_pin
```
[the remark this answers is not in the document]
Comment (davidos-b) [reply include/ardpolyfills/Servo.h:32]: So no doc for this one?

File: include/ardpolyfills/Servo.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (RuthgerD) [reply include/ardpolyfills/Servo.h:47]: Accoddingly

File: include/ardpolyfills/Servo.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (RuthgerD) [reply include/ardpolyfills/Servo.h:32]: It should explain what it does for the user not what it does internally

File: include/ardpolyfills/Servo.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidos-b) [reply include/ardpolyfills/Servo.h:47]: Ooohhh

**Comment by RuthgerD** [comment]
please fix odd git history

**Pull request closed without merging**